File: 3-Repetitions/repetitions.js
const test1 = "ATTCGGGA";

function maxRepetitions(str) {
  if (str.length === 0) {
    return 0;
  } else if (str.length === 1) {
    return 1;
  }

  let maxLength = 1;
  let currentLength = 1;

  for (let i = 1; i < str.length; i++) {
    if (str[i] === str[i - 1]) {
      currentLength++;
    } else {
      currentLength = 1;
    }

    maxLength = Math.max(maxLength, currentLength);
  }

  return maxLength;
}

console.log(maxRepetitions(test1));

File: 3-Repetitions/repetitions.py
test_1 = "ATTCGGGA";

def max_repetitions(str):
	if len(str) == 0:
		return 0
	elif len(str) == 1:
		return 1
	
	max_length = 1
	current_length = 1
	
	current_char = str[0]

	for char in str[1:]:
		if char == current_char:
			current_length += 1
			if current_length > max_length:
				max_length = current_length
		else:
			current_char = char
			current_length = 1 
		
	return max_length
        

print(max_repetitions(test_1))